src/Symfony/Component/Message/MessageConsumerInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+interface MessageConsumerInterface
+{
+    public function consume(): \Generator;
```

Comment:
I'm really not found of the `messages()` suggestion but agree the naming can be better. What about `FetcherInterface` that has a `fetch(): \Generator` method? Because these things are fetching messages from various sources...

Reply:
Or `FetchedMessage::messages()` :) but `Fetcher::fetch()` is more clear than the previous proposal.